Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM python:3.9.0-buster as python-base
+WORKDIR /app
+# RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+# RUN cd /project
+# #RUN  source #!usr/bin/env/
+# RUN $HOME/.poetry/env 
+# RUN poetry install
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_NO_CACHE_DIR=off \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    PIP_DEFAULT_TIMEOUT=100 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 
+    
+
+ENV PATH="$POETRY_HOME/bin:$PATH"
+
+# builder-base is used to build dependencies
+FROM python-base as builder-base
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        curl \
+        build-essential
+
+# Install Poetry - respects $POETRY_VERSION & $POETRY_HOME
+ENV POETRY_VERSION=1.1.4
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+# We copy our Python requirements here to cache them
+# and install only runtime deps using poetry
+
+COPY ./poetry.lock ./pyproject.toml ./
+
+# Use below -To run gunicorn with poetry
+ENTRYPOINT [ "poetry", "run", "gunicorn", "--bind", "0.0.0.0:5000", "todo_app.wsgi:app" ]
```

Comment:
You can remove this line - you have an ENTRYPOINT for each of the child stages so this one always gets replaced.